src/backend/api/Fusion.Resources.Domain/Queries/GetDelegatedDepartmentResponsibles.cs

```diff
@@ -37,17 +44,20 @@ public Handler(ResourcesDbContext db, IFusionProfileResolver profileResolver, IM
             public async Task<IEnumerable<QueryDepartmentResponsible>> Handle(GetDelegatedDepartmentResponsibles request, CancellationToken cancellationToken)
             {
                 var returnModel = new List<QueryDepartmentResponsible>();
-
                 var department = await mediator.Send(new GetDepartment(request.DepartmentId));
-
                 if (department is null)
                     return returnModel;
 
+
                 var delegatedResourceOwners = await db.DelegatedDepartmentResponsibles
```

Comment:
You could split this into a query.
example:
var query = db.TableName.AsNoTracking().Where(x=>x.Dep=="whatever");
if(haveDateCriteria){
query = query.Where(x=>x.Date => DateTimeOffset.UtcNow.Date);
}
var myEndResult = query.ToListAsync(cancellationtoken);

As of now you first issue the query against database, then do a date filter after you fetched the data.